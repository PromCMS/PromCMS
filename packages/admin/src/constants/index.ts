export * from './adminLanguages';
export * from './api';
export * from './localization';
export * from './pageUrls';
export * from './misc';
export * from './texts';
