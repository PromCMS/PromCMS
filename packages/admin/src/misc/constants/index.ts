export * from './api'
export * from './localization'
export * from './texts'
export * from './misc'
