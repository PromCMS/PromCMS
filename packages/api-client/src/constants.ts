export const HTTP_STATUS_CODE_FILE_TOO_LARGE = 413;
export const HTTP_STATUS_CODE_UNSUPPORTED_FILE_EXTENSION = 415;
