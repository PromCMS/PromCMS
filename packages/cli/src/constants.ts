import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

export const PACKAGE_ROOT = path.join(__dirname, '..');
export const SCRIPTS_ROOT = path.join(PACKAGE_ROOT, 'scripts');
export const PROJECT_ROOT = path.join(PACKAGE_ROOT, '..', '..');
export const TEMPLATES_ROOT = path.join(PACKAGE_ROOT, 'templates');
