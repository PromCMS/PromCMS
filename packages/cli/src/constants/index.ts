import { join } from 'path';
import { dirname } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));

export const PROJECT_ROOT = join(__dirname, '..', '..', '..');

export const CORE_ROOT = join(PROJECT_ROOT, 'packages', 'core');
export const CLI_ROOT = join(PROJECT_ROOT, 'packages', 'cli');
