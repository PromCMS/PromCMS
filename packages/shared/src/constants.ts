export const supportedConfigExtensions = [
  'json',
  'js',
  'cjs',
  'mjs',
  'ts',
] as const;
