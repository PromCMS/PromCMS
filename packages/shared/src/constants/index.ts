import findConfig from 'find-config';
import { join } from 'path';
import { dirname } from 'path';
import { ExportConfig } from '../types';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

export const PROJECT_ROOT = join(__dirname, '..', '..', '..', '..');

export const ADMIN_PROJECT_ROOT = join(PROJECT_ROOT, 'packages', 'admin');
export const CORE_ROOT = join(PROJECT_ROOT, 'packages', 'core');

export const GENERATOR_FILENAME = 'prom.generate-config.js';
export const GENERATOR_FILE_PATH = join(PROJECT_ROOT, GENERATOR_FILENAME);
export const GENERATOR_CONFIG: ExportConfig | undefined =
  findConfig.require(GENERATOR_FILENAME, { module: true }) ||
  JSON.parse(process.env.PROM_JSON_CONFIG || 'null');
