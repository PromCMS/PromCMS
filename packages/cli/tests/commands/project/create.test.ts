import { execa } from 'execa';
import path from 'path';
import fs from 'fs-extra';
import { describe, beforeEach, it, afterAll, expect } from 'vitest';

import { monorepoRoot } from '@prom-cms/shared/dist/internal/constants.js';

const TEST_FOLDER_PATH = path.join(
  monorepoRoot,
  'node_modules',
  '.prom-cms',
  'test-app'
);
describe('commands', () => {
  describe(
    'project create',
    () => {
      beforeEach(async () => {
        await fs.remove(TEST_FOLDER_PATH);
        await fs.ensureDir(TEST_FOLDER_PATH);
      });

      afterAll(async () => {
        await fs.remove(TEST_FOLDER_PATH);
      });

      it('should run correctly and generate files', async () => {
        await expect(
          execa(
            'node',
            [
              path.join(monorepoRoot, 'packages/cli/bin/cli.js'),
              'project',
              'create',
              '--packageManager',
              'npm',
              '--name',
              'Test Project',
              '--no-install',
            ],
            {
              cwd: TEST_FOLDER_PATH,
            }
          )
        ).resolves.to.not.toThrow();
      });
    },
    {
      timeout: 2 * 60 * 1000, // 1 minute
    }
  );
});
