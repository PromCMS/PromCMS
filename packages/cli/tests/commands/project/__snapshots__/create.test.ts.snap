// Vitest Snapshot v1

exports[`commands > project create > should run correctly and generate files > /.env 1`] = `
"APP_NAME=\\"Test Project\\"
APP_ENV=\\"production\\"
APP_PREFIX=
APP_DEBUG=true
APP_URL=\\"http://localhost\\"
MAIL_HOST=\\"\\"
MAIL_PORT=
MAIL_USER=\\"\\"
MAIL_PASS=\\"\\"
MAIL_ADDRESS=\\"\\"
SECURITY_SECRET=\\"f5ee3511776897d6712c7d7f839fabcf9df6bd82\\"
"
`;

exports[`commands > project create > should run correctly and generate files > /.gitignore.md 1`] = `
".temp
cache
uploads
vendor
.database
node_modules
*.env*
_.*
"
`;

exports[`commands > project create > should run correctly and generate files > /.htaccess 1`] = `
"RewriteEngine On
RewriteBase /
ErrorDocument 404 /public/index.php
RewriteCond %{REQUEST_URI} !^/public
RewriteRule (.*) public/$1 [L,QSA]"
`;

exports[`commands > project create > should run correctly and generate files > /.prettierrc.json 1`] = `
"{
  \\"trailingComma\\": \\"es5\\",
  \\"tabWidth\\": 2,
  \\"semi\\": false,
  \\"singleQuote\\": true,
  \\"phpVersion\\": \\"8.2\\",
  \\"plugins\\": [
    \\"./node_modules/prettier-plugin-twig-melody\\",
    \\"./node_modules/@prettier/plugin-php\\"
  ]
}
"
`;

exports[`commands > project create > should run correctly and generate files > /.prom-cms/.htaccess 1`] = `"deny from all"`;

exports[`commands > project create > should run correctly and generate files > /.prom-cms/config.json 1`] = `
"{
  \\"project\\": {
    \\"name\\": \\"Test Project\\",
    \\"slug\\": \\"test-project\\",
    \\"url\\": \\"http://localhost\\"
  },
  \\"database\\": {
    \\"connections\\": [
      {
        \\"name\\": \\"default-connection\\",
        \\"adapter\\": \\"sqlite\\",
        \\"dsn\\": \\"sqlite:/Users/daichi/Desktop/Work/prom-cms/prom/node_modules/.prom-cms/test-app/.database/application.sq3\\",
        \\"user\\": \\"root\\",
        \\"password\\": null
      }
    ]
  }
}
"
`;

exports[`commands > project create > should run correctly and generate files > /.prom-cms/propel/config/config.php 1`] = `
"<?php
$serviceContainer = \\\\Propel\\\\Runtime\\\\Propel::getServiceContainer();
$serviceContainer->checkVersion(2);
$serviceContainer->setAdapterClass('default-connection', 'sqlite');
$manager = new \\\\Propel\\\\Runtime\\\\Connection\\\\ConnectionManagerSingle('default-connection');
$manager->setConfiguration(array (
  'dsn' => 'sqlite:/Users/daichi/Desktop/Work/prom-cms/prom/node_modules/.prom-cms/test-app/.database/application.sq3',
  'user' => 'root',
  'password' => '',
  'settings' =>
  array (
    'charset' => 'utf8',
    'queries' =>
    array (
    ),
  ),
  'classname' => '\\\\\\\\Propel\\\\\\\\Runtime\\\\\\\\Connection\\\\\\\\ConnectionWrapper',
  'model_paths' =>
  array (
    0 => 'src',
    1 => 'vendor',
  ),
));
$serviceContainer->setConnectionManager($manager);
$serviceContainer->setDefaultDatasource('default-connection');
require_once __DIR__ . '/./loadDatabase.php';
"
`;

exports[`commands > project create > should run correctly and generate files > /.prom-cms/propel/config/loadDatabase.php 1`] = `
"<?php
$serviceContainer = \\\\Propel\\\\Runtime\\\\Propel::getServiceContainer();
$serviceContainer->initDatabaseMapFromDumps(array (
  'default-connection' => 
  array (
    'tablesByName' => 
    array (
    ),
    'tablesByPhpName' => 
    array (
    ),
  ),
));
"
`;

exports[`commands > project create > should run correctly and generate files > /.prom-cms/propel/propel.json 1`] = `
"{
  \\"propel\\": {
    \\"paths\\": {
      \\"schemaDir\\": \\"./.prom-cms/propel\\",
      \\"phpDir\\": \\"./modules/Test/Models\\",
      \\"phpConfDir\\": \\"./.prom-cms/propel/config\\",
      \\"migrationDir\\": \\"./.prom-cms/propel/migrations\\",
      \\"sqlDir\\": \\"./.prom-cms/propel/sql\\"
    },
    \\"generator\\": {
      \\"namespaceAutoPackage\\": false
    },
    \\"database\\": {
      \\"connections\\": {
        \\"default-connection\\": {
          \\"adapter\\": \\"sqlite\\",
          \\"dsn\\": \\"sqlite:/Users/daichi/Desktop/Work/prom-cms/prom/node_modules/.prom-cms/test-app/.database/application.sq3\\",
          \\"user\\": \\"root\\",
          \\"password\\": null,
          \\"settings\\": {
            \\"charset\\": \\"utf8\\"
          }
        }
      }
    }
  }
}
"
`;

exports[`commands > project create > should run correctly and generate files > /.prom-cms/propel/schema.xml 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<database
  name=\\"default-connection\\"
  defaultIdMethod=\\"native\\"
  namespace=\\"PromCMS\\\\Modules\\\\Test\\\\Models\\"
>
</database>
"
`;

exports[`commands > project create > should run correctly and generate files > /README.md 1`] = `
"# \`PromCMS project\`
> TODO: description
"
`;

exports[`commands > project create > should run correctly and generate files > /composer.json 1`] = `
"{
  \\"name\\": \\"prom-cms/test-project\\",
  \\"type\\": \\"project\\",
  \\"authors\\": [],
  \\"require\\": {
    \\"prom-cms/core\\": \\"dev-develop\\"
  },
  \\"minimum-stability\\": \\"dev\\",
  \\"autoload\\": {
    \\"psr-4\\": {
      \\"PromCMS\\\\\\\\Modules\\\\\\\\\\": \\"modules/\\"
    }
  },
  \\"require-dev\\": {
    \\"fakerphp/faker\\": \\"2.0.x-dev\\",
    \\"phpunit/phpunit\\": \\"^11.0@dev\\"
  }
}
"
`;

exports[`commands > project create > should run correctly and generate files > /docker-compose.yml 1`] = `
"version: '3'
services:
  php_web:
    container_name: test-project_web
    build:
      context: '.'
      dockerfile: Dockerfile
    env_file:
      - '.env'
    volumes:
      - './:/var/www/html'
    user: '1000:1000'
    ports:
      - '80:80'
"
`;

exports[`commands > project create > should run correctly and generate files > /frontend-src/.htaccess 1`] = `"deny from all"`;

exports[`commands > project create > should run correctly and generate files > /frontend-src/index.ts 1`] = `
"console.log('Hello there');
"
`;

exports[`commands > project create > should run correctly and generate files > /models/.htaccess 1`] = `"deny from all"`;

exports[`commands > project create > should run correctly and generate files > /modules/Test/Views/components/BlocksRenderer.twig 1`] = `
"<div class=\\"block-renderer\\"> 
  {% if blocks %}
        {% for block in blocks %}
          {% if block.type == 'header' %}
            <h{{ block.data.level|default(2) }}>
              {{ block.data.text|raw }}
            </h{{ block.data.level|default(2) }}>
          {% elseif block.type == 'table' %}
            <table class=\\"table\\"> 
              {% if block.data.withHeadings %}
                <thead> 
                  <tr> 
                    {% for cell in block.data.content[0] %}
                      <th scope=\\"col\\">{{cell|raw}}</td> 
                    {% endfor %}
                  </tr>
                </thead>
              {% endif %}
              <tbody>
                {% if block.data.withHeadings %}
                  {% for row in block.data.content[1:] %}
                    <tr> 
                        {% for cell in row %}
                          <td>{{cell|raw}}</td> 
                        {% endfor %}
                    </tr>
                  {% endfor %}
                {% else %}
                  {% for row in block.data.content %}
                    <tr> 
                        {% for cell in row %}
                          <td>{{cell|raw}}</td> 
                        {% endfor %}
                    </tr>
                  {% endfor %}
                {% endif %}
              </tbody>
            </table>
          {% elseif block.type == 'paragraph' %}
            <p>
              {{ block.data.text }}
            </p>
          {% elseif block.type == 'image' %}
            {% set fullImage = image(block.data.fileId, 1240) %}
            {% set thumbnail = image(block.data.fileId, 1240, null, 85) %}
            <div class=\\"big-image w-full gallery-photoswipe\\">
              <a class=\\"w-full\\" href=\\"{{ fullImage.src }}\\" data-pswp-width=\\"{{fullImage.width}}\\" data-pswp-height=\\"{{fullImage.height}}\\">
                  <img src=\\"{{thumbnail.src}}\\" width=\\"{{thumbnail.width}}\\" height=\\"{{thumbnail.height}}\\" alt=\\"{{ block.data.label }}\\" />
                  <div class=\\"d-none hidden-caption-content\\">
                    {% if block.data.label %}
                        <p class=\\"heading\\">{{block.data.label}}</p>
                    {% endif %}
                    {% if block.data.description %}
                        <p class=\\"description\\">{{block.data.description}}</p>
                    {% endif %}
                  </div>
              </a>
            </div>
          {% elseif block.type == 'big-gallery' %}
             <div class=\\"tabs\\">
              {% for gallery in block.data %}
                <input type=\\"radio\\"
                  name=\\"tabs\\"
                  id=\\"tab{{ loop.index }}\\"
                  checked=\\"checked\\" />
                <label for=\\"tab{{ loop.index }}\\">
                  {% if gallery.data.label %}
                    {{ gallery.data.label }}
                  {% else %}
                    Žádný popisek
                  {% endif %}
                </label>
                <div class=\\"tab gallery-photoswipe portfolio-gallery\\">
                  {% for file in gallery.data.fileIds %}
                    {% set fullImage = image(file.id, 1240) %}
                    {% set thumbnail = image(file.id, 400, null, 60) %}
                    <a href=\\"{{ fullImage.src }}\\"
                      class=\\"gallery-item\\"
                      data-pswp-width=\\"{{ fullImage.width }}\\"
                      data-pswp-height=\\"{{ fullImage.height }}\\">
                      <img src=\\"{{ thumbnail.src }}\\"
                        width=\\"{{ thumbnail.width }}\\"
                        alt=\\"{{ file.title }}\\" />
                      <div class=\\"d-none hidden-caption-content\\">
                        {% if file.title %}
                          <p class=\\"heading\\">
                            {{ file.title }}
                          </p>
                        {% endif %}
                        {% if file.description %}
                          <p class=\\"description\\">
                            {{ file.description }}
                          </p>
                        {% endif %}
                      </div>
                    </a>
                  {% endfor %}
                </div>
              {% endfor %}
            </div>
          {% elseif block.type == 'gallery' %}
            <div class=\\"gallery gallery-photoswipe\\">
              {% for file in block.data.fileIds %}
                {% set fullImage = image(file.id, 1240) %}
                {% set thumbnail = image(file.id, 400, null, 60) %}
                <a href=\\"{{ fullImage.src }}\\"
                  data-pswp-width=\\"{{ fullImage.width }}\\"
                  data-pswp-height=\\"{{ fullImage.height }}\\">
                  <img src=\\"{{ thumbnail.src }}\\"
                    width=\\"{{ thumbnail.width }}\\"
                    alt=\\"{{ file.title }}\\" />
                  <div class=\\"d-none hidden-caption-content\\">
                    {% if file.title %}
                      <p class=\\"heading\\">
                        {{ file.title }}
                      </p>
                    {% endif %}
                    {% if file.description %}
                      <p class=\\"description\\">
                        {{ file.description }}
                      </p>
                    {% endif %}
                  </div>
                </a>
              {% endfor %}
            </div>
          {% elseif block.type == 'list' %}
            <ul>
              {% for listItem in block.data.items %}
                <li>{{ listItem|raw }}</li>
              {% endfor %}
            </ul>
          {% elseif block.type == 'dynamicBlock' %}
            {{ getDynamicBlock(block.data.blockId, data)|raw }}
          {% elseif block.type == 'tags' %}
            <ul class=\\"badges\\">
              {% for tag in block.data.tags %}
                <li>{{ tag|raw }}</li>
              {% endfor %}
            </ul>
          {% elseif block.type == 'buttonLink' %}
            <div class=\\"btn-box {% if block.data.placeholderImage %}is-image{% endif %}\\">
              <a href=\\"{{block.data.linkTo}}?q=90&w=1240\\" title=\\"{{block.data.label}}\\" {% if block.data.openOnNewTab %} target=\\"_blank\\" {% endif %} {% if block.data.isDownload %} download {% endif %}>
                {% if block.data.placeholderImage %}
                  <img class=\\"big-image w-100\\" src=\\"/api/entry-types/files/items/{{ block.data.placeholderImage }}/raw?q=70&w=400\\" />
                {% else %}
                  {% if block.data.isDownload %}
                    <svg xmlns=\\"http://www.w3.org/2000/svg\\"
                      x=\\"0px\\"
                      y=\\"0px\\"
                      width=\\"20\\"
                      height=\\"20\\"
                      viewBox=\\"0 0 172 172\\"
                      style=\\" fill:#000000;\\">
                      <g fill=\\"none\\"
                        fill-rule=\\"nonzero\\"
                        stroke=\\"none\\"
                        stroke-width=\\"1\\"
                        stroke-linecap=\\"butt\\"
                        stroke-linejoin=\\"miter\\"
                        stroke-miterlimit=\\"10\\"
                        stroke-dasharray=\\"\\"
                        stroke-dashoffset=\\"0\\"
                        font-family=\\"none\\"
                        font-weight=\\"none\\"
                        font-size=\\"none\\"
                        text-anchor=\\"none\\"
                        style=\\"mix-blend-mode: normal\\">
                        <path d=\\"M0,172v-172h172v172z\\" fill=\\"none\\"></path>
                        <g fill=\\"#fff\\">
                          <path d=\\"M78.83333,14.33333c-3.956,0 -7.16667,3.21067 -7.16667,7.16667v57.33333h-25.08333c-1.97902,0 -3.58333,1.60431 -3.58333,3.58333c0.00014,0.95031 0.37777,1.86164 1.0498,2.53353c0.07162,0.07315 0.14634,0.1432 0.22396,0.20995l36.58919,34.50359l0.23795,0.22396c1.32656,1.24737 3.07819,1.94302 4.89909,1.94564c1.8209,-0.00261 3.57253,-0.69826 4.89909,-1.94564l0.09798,-0.08399c0.02354,-0.02782 0.04687,-0.05581 0.06998,-0.08399l36.5752,-34.48958c0.07258,-0.06249 0.14262,-0.12786 0.20996,-0.19597l0.05599,-0.04199c0.01411,-0.01388 0.02811,-0.02788 0.042,-0.04199c0.67204,-0.6719 1.04966,-1.58323 1.0498,-2.53353c0,-1.97902 -1.60431,-3.58333 -3.58333,-3.58333h-25.08333v-57.33333c0,-3.956 -3.21067,-7.16667 -7.16667,-7.16667h-7.16667zM21.5,143.33333c-2.58456,-0.03655 -4.98858,1.32136 -6.29153,3.55376c-1.30295,2.2324 -1.30295,4.99342 0,7.22582c1.30295,2.2324 3.70697,3.59031 6.29153,3.55376h129c2.58456,0.03655 4.98858,-1.32136 6.29153,-3.55376c1.30295,-2.2324 1.30295,-4.99342 0,-7.22582c-1.30295,-2.2324 -3.70697,-3.59031 -6.29153,-3.55376z\\"></path>
                        </g>
                      </g>
                    </svg>
                  {% endif %}
                  {% if block.data.label %}
                    {{ block.data.label|raw }}
                  {% else %}
                    {{block.data.linkTo}}
                  {% endif %}
                {% endif %}
              </a>
            </div>
          {% elseif block.type == 'columns' %}
            {% set filteredCols = block.data.itemContent|filter( v => v.blocks|length > 0 ) %}
            {% set size = filteredCols|length %}
            {% if size is same as(1) %}
              {% set colsWrapperClassNames = \\"col-12\\" %}
            {% elseif size is same as(2) %}
              {% set colsWrapperClassNames = \\"col-12 col-sm-6\\" %}
            {% elseif size is same as(3) %}
              {% set colsWrapperClassNames = \\"col-12 col-md-4\\" %}
            {% else %}
              {% set colsWrapperClassNames = \\"col-12 col-sm-6 col-md-3\\" %}
            {% endif %}
            {% if size is same as(1) %}
              {% set gutterClassName = \\"gx-0 gy-0\\" %}
            {% elseif size is same as(2) %}
              {% set gutterClassName = \\"gx-5 gy-0\\" %}
            {% else %}
              {% set gutterClassName = \\"gx-5 gy-0\\" %}
            {% endif %}
            <div class=\\"row {{gutterClassName}}\\">
              {% for col in filteredCols %}
                <div class=\\"{{colsWrapperClassNames}}\\">
                  {% include 'components/BlocksRenderer.twig' with {blocks: col.blocks} %}
                </div>
              {% endfor %}
            </div>
          {% elseif block.type == 'embed' %}
            {% if block.data.service == 'youtube' %}
              <iframe width=\\"100%\\"
                height=\\"515\\"
                src=\\"{{ block.data.embed }}\\"
                title=\\"YouTube video player\\"
                frameborder=\\"0\\"
                allow=\\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\\"
                allowfullscreen></iframe>
            {% endif %}
          {% endif %}
        {% endfor %}
      {% endif %}
</div>"
`;

exports[`commands > project create > should run correctly and generate files > /modules/Test/Views/layouts/site-layout/footer.twig 1`] = `
"<footer>Hello, this is footer 👋🏻</footer>;
"
`;

exports[`commands > project create > should run correctly and generate files > /modules/Test/Views/layouts/site-layout/header.twig 1`] = `
"<header>Hello, this is header 👋🏻</header>;
"
`;

exports[`commands > project create > should run correctly and generate files > /modules/Test/Views/layouts/site-layout/site-layout.twig 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\" />
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\" />
    <meta name=\\"viewport\\"
      content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\" />
    {# Info #}
    <title>
      {{ block('title') }} - Test Project
    </title>
    {# SEO #}
    <meta name=\\"description\\" content=\\"Test Project\\" />
    {# Assets #}
    {{
      getViteAssets({
        distFolderPath: 'dist',
        assets: [
          {
            path: 'index.ts',
            type: 'script'
          }
        ]
      })|raw
    }}
  </head>
  <body>
    {% include '@modules:Test/layouts/site-layout/header.twig' %}
    <main>
      {{ block('content') }}
    </main>
    {% include '@modules:Test/layouts/site-layout/footer.twig' %}
  </body>
</html>
"
`;

exports[`commands > project create > should run correctly and generate files > /modules/Test/Views/pages/home.twig 1`] = `
"{% extends '@modules:Test/layouts/site-layout/site-layout.twig' %}
{% block title %}
  Home page
{% endblock %}
{% block content %}
  Hello from homepage 🎉
{% endblock %}
"
`;

exports[`commands > project create > should run correctly and generate files > /modules/Test/bootstrap.php 1`] = `
"<?php
// In this file you can tell what this module contains or have here something that should be loaded before your models, routes, ..etc
use Slim\\\\App;

return function (App $app) {
  // You can access container and other stuff like this
  // $container = $app->getContainer();
};
"
`;

exports[`commands > project create > should run correctly and generate files > /modules/Test/front.routes.php 1`] = `
"<?php
use Psr\\\\Http\\\\Message\\\\ResponseInterface;
use Psr\\\\Http\\\\Message\\\\ServerRequestInterface;
use Slim\\\\App;
use Slim\\\\Views\\\\Twig as TwigViews;
use Slim\\\\Routing\\\\RouteCollectorProxy;
return function (App $app, RouteCollectorProxy $router) {
  $router->get('/', function (
    ServerRequestInterface $request,
    ResponseInterface $response,
    $args,
  ) use ($app) {
    $container = $app->getContainer();
    $twig = $container->get(TwigViews::class);
    return $twig->render($response, '@modules:Test/pages/home.twig');
  });
};
"
`;

exports[`commands > project create > should run correctly and generate files > /modules/Test/module-info.json 1`] = `
"{
  \\"name\\": \\"Test\\"
}
"
`;

exports[`commands > project create > should run correctly and generate files > /package.json 1`] = `
"{
  \\"name\\": \\"test-project\\",
  \\"version\\": \\"0.0.0\\",
  \\"description\\": \\"\\",
  \\"license\\": \\"MIT\\",
  \\"type\\": \\"module\\",
  \\"scripts\\": {
    \\"build\\": \\"vite build\\",
    \\"dev\\": \\"vite\\"
  },
  \\"devDependencies\\": {
    \\"@prettier/plugin-php\\": \\"^0.21.0\\",
    \\"@prom-cms/vite-plugin\\": \\"^0.7.1\\",
    \\"prettier-plugin-twig-melody\\": \\"^0.4.6\\",
    \\"typescript\\": \\"^5.3.3\\",
    \\"vite\\": \\"^4.5.1\\",
    \\"vite-plugin-live-reload\\": \\"^3.0.3\\"
  }
}
"
`;

exports[`commands > project create > should run correctly and generate files > /private/.htaccess 1`] = `"deny from all"`;

exports[`commands > project create > should run correctly and generate files > /public/.htaccess 1`] = `
"<ifModule mod_expires.c> 
  ExpiresActive On 
  ExpiresByType image/x-icon \\"access plus 600 seconds\\" 
  ExpiresByType image/jpeg \\"access plus 600 seconds\\" 
  ExpiresByType image/jpg \\"access plus 600 seconds\\" 
  ExpiresByType image/png \\"access plus 600 seconds\\" 
  ExpiresByType image/gif \\"access plus 600 seconds\\" 
  ExpiresByType application/x-shockwave-flash \\"access plus 600 seconds\\" 
  ExpiresByType text/css \\"access plus 600 seconds\\" 
  ExpiresByType text/javascript \\"access plus 600 seconds\\" 
  ExpiresByType application/javascript \\"access plus 600 seconds\\" 
  ExpiresByType application/x-javascript \\"access plus 600 seconds\\" 
</ifModule> 
<ifModule mod_rewrite.c> 
  RewriteEngine On
  RewriteBase /
  # HTTP >> HTTPS 
  RewriteCond %{HTTPS} !=on
  RewriteCond %{HTTP_HOST} !=localhost:3001
  RewriteRule ^ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
  ## non www to www
  RewriteCond %{HTTP_HOST} !=localhost:3001
  RewriteCond %{HTTP_HOST} !^test\\\\.
  RewriteCond %{HTTP_HOST} !^www\\\\.
  RewriteRule ^(.*)$ http://www.%{HTTP_HOST}/$1 [L,R=301]
  # Server only public folder to public
  RewriteCond %{REQUEST_URI} !^/public/
  RewriteRule (.*) public/$1
  # If we did not find any file then we handle requests with php app
  RewriteCond %{REQUEST_FILENAME} !-f
  RewriteRule (.*) public/index.php
</ifModule> 
"
`;

exports[`commands > project create > should run correctly and generate files > /public/admin/index.html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <meta name=\\"description\\" content=\\"Another PromCMS project\\" />
    <meta name=\\"robots\\" content=\\"nofollow,noindex\\" />
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"/admin/assets/logo-9f27b0a7.ico\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <link rel=\\"preconnect\\" href=\\"https://fonts.googleapis.com\\" />
    <link rel=\\"preconnect\\" href=\\"https://fonts.gstatic.com\\" />
    <link
      href=\\"https://fonts.googleapis.com/css2?family=Open+Sans:ital,wght@0,300;0,400;0,500;0,600;0,700;1,300;1,400;1,500&display=swap\\"
      rel=\\"stylesheet\\" 
    />
    <title>PromCMS</title>
    <script type=\\"module\\" crossorigin src=\\"/admin/assets/index-d65991cf.js\\"></script>
    <link rel=\\"stylesheet\\" href=\\"/admin/assets/index-41e8caf1.css\\">
  </head>
  <body>
    <div id=\\"prom_cms_root\\">
      <style>body{margin:0}.prom-cms-loader{align-items:center;background-color:#eff6ff;display:flex;height:100vh;justify-content:center;width:100vw}.prom-cms-loader>div{position:relative}@keyframes infiniteLoader{0%{left:-20%;width:50%}to{left:120%;width:20%}0%,to{transform:translateX(-50%)}}.prom-cms-loader__text{display:block;margin-bottom:.6rem;text-align:center}.prom-cms-loader__spinner{background-color:#fff;border-radius:9999px;display:block;height:.7rem;min-width:300px;overflow:hidden;position:relative}.prom-cms-loader__spinner:after{animation:infiniteLoader 1.5s ease-out infinite;background-color:#60a5fa;border-radius:9999px;content:\\"\\";display:block;height:100%;position:absolute;top:0}</style>
      <div class=\\"prom-cms-loader\\">
        <div>
          <span class=\\"prom-cms-loader__text\\">Booting up...</span>
          <span class=\\"prom-cms-loader__spinner\\" />
        </div>
      </div>
    </div>
    
  </body>
</html>"
`;

exports[`commands > project create > should run correctly and generate files > /public/index.php 1`] = `
"<?php
use PromCMS\\\\Core\\\\Path;
include_once __DIR__ . '/../vendor/autoload.php';
$app = new \\\\PromCMS\\\\Core\\\\App(Path::join(__DIR__, '..'));
$app->init();
$app->run();
"
`;

exports[`commands > project create > should run correctly and generate files > /tsconfig.json 1`] = `
"{
  \\"compilerOptions\\": {
    \\"declaration\\": true,
    \\"noImplicitAny\\": false,
    \\"removeComments\\": true,
    \\"noLib\\": false,
    \\"esModuleInterop\\": true,
    \\"emitDecoratorMetadata\\": true,
    \\"experimentalDecorators\\": true,
    \\"sourceMap\\": true,
    \\"target\\": \\"ESNext\\",
    \\"lib\\": [\\"dom\\", \\"dom.iterable\\", \\"esnext\\", \\"es6\\"],
    \\"allowJs\\": true,
    \\"skipLibCheck\\": true,
    \\"strict\\": true,
    \\"forceConsistentCasingInFileNames\\": true,
    \\"strictPropertyInitialization\\": false,
    \\"noEmit\\": false,
    \\"module\\": \\"ESNext\\",
    \\"moduleResolution\\": \\"node\\",
    \\"resolveJsonModule\\": true,
    \\"isolatedModules\\": false,
    \\"jsx\\": \\"react-jsx\\",
    \\"incremental\\": true,
    \\"useDefineForClassFields\\": true,
    \\"allowSyntheticDefaultImports\\": true,
    \\"outDir\\": \\"public/bundle\\"
  },
  \\"include\\": [\\"./frontend-src\\"],
  \\"exclude\\": [\\"node_modules\\", \\"public\\"]
}
"
`;

exports[`commands > project create > should run correctly and generate files > /vite.config.ts 1`] = `
"import { defineConfig } from 'vite';
import liveReload from 'vite-plugin-live-reload';
import { promCmsVitePlugin } from '@prom-cms/vite-plugin';
export default defineConfig({
  plugins: [
    liveReload('../(modules|public)/**/*.(php|ts|js|css|scss|json|twig)'),
    promCmsVitePlugin(),
  ],
});
"
`;
